### PR TITLE
Use DSS parameter store to determine bucket names (#1314)

### PR DESCRIPTION
**Pull request by achave11**

### CODE REVIEW

File: src/azul/__init__.py [hunk 1]
```diff
@@ -155,13 +159,13 @@ def dss_checkout_bucket(self, dss_endpoint: Optional[str] = None):
...
+        ssm = boto3.client('ssm')
         stage = self._dss_deployment_stage(dss_endpoint)
-        # For domain_part, DSS went from `humancellatlas` to `hca` in 9/2018 and started reverting back to
-        # `humancellatlas` in 12/2018. As I write this, only `dev` is back on `humancellatlas`
-        domain_part = 'hca' if stage == 'prod' else 'humancellatlas'
-        qualifier = [qualifier] if qualifier else []
-        return '-'.join(['org', domain_part, 'dss', *qualifier, stage])
+        name = f'dcp/dss/{stage}/environment'
+        dss_config = json.loads(ssm.get_parameter(Name=name)['Parameter']['Value'])
+        return dss_config['DSS_S3_BUCKET'] if not qualifier else dss_config['DSS_S3_CHECKOUT_BUCKET']
```
Comment (hannes-ucsc) [inline src/azul/__init__.py]: This is very brittle. What if we add a third value for qualifier? Which values for the qualifier argument do you currently support: None and 'checkout'. Any other value should raise an exception. 

Alternatively, you could change the type of that argument so it really only supports the two cases, normal bucket or checkout bucket.

BUT: Note how the qualifier value actually occurs in the name of the variable. Just in uppercase. Study the old logic and adjust it.

File: src/azul/__init__.py [hunk 2]
```diff
@@ -155,13 +159,17 @@ def dss_checkout_bucket(self, dss_endpoint: Optional[str] = None):
...
+        ssm = boto3.client('ssm')
         stage = self._dss_deployment_stage(dss_endpoint)
-        # For domain_part, DSS went from `humancellatlas` to `hca` in 9/2018 and started reverting back to
-        # `humancellatlas` in 12/2018. As I write this, only `dev` is back on `humancellatlas`
-        domain_part = 'hca' if stage == 'prod' else 'humancellatlas'
+        name = f'dcp/dss/{stage}/environment'
+        dss_config = json.loads(ssm.get_parameter(Name=name)['Parameter']['Value'])
         qualifier = [qualifier] if qualifier else []
-        return '-'.join(['org', domain_part, 'dss', *qualifier, stage])
+        bucket_key = '_'.join(['dss_s3', *qualifier, "bucket"]).upper()
```
Comment (hannes-ucsc) [inline src/azul/__init__.py]:
```suggestion
        bucket_key = '_'.join(['dss', 's3', *qualifier, 'bucket']).upper()
```

**Comment by achave11** [comment]
I know what it is. Addressing it now.

File: src/azul/__init__.py [hunk 3]
```diff
@@ -155,13 +159,18 @@ def dss_checkout_bucket(self, dss_endpoint: Optional[str] = None):
...
         stage = self._dss_deployment_stage(dss_endpoint)
-        # For domain_part, DSS went from `humancellatlas` to `hca` in 9/2018 and started reverting back to
-        # `humancellatlas` in 12/2018. As I write this, only `dev` is back on `humancellatlas`
-        domain_part = 'hca' if stage == 'prod' else 'humancellatlas'
+        name = f'/dcp/dss/{stage}/environment'
+        dss_parameter = ssm.get_parameter(Name=name)
+        dss_config = _json.loads(dss_parameter['Parameter']['Value'])
         qualifier = [qualifier] if qualifier else []
-        return '-'.join(['org', domain_part, 'dss', *qualifier, stage])
+        bucket_key = '_'.join(['dss', 's3', *qualifier, "bucket"]).upper()
```
Comment (hannes-ucsc) [inline src/azul/__init__.py]:
```suggestion
        bucket_key = '_'.join(['dss', 's3', *qualifier, 'bucket']).upper()
```

File: README.md [hunk 4]
```diff
@@ -492,6 +492,39 @@ Error inspecting states in the "s3" backend:
...
+  File "/var/task/azul/dss.py", line 49, in __init__
+    self.bucket = config.dss_main_bucket(dss_endpoint)
+  File "/var/task/azul/__init__.py", line 178, in dss_main_bucket
+    return self._dss_bucket(dss_endpoint)
+  File "/var/task/azul/__init__.py", line 169, in _dss_bucket
+    dss_parameter = ssm.get_parameter(Name=name)
+  File "/var/task/botocore/client.py", line 314, in _api_call
+    return self._make_api_call(operation_name, kwargs)
+  File "/var/task/botocore/client.py", line 612, in _make_api_call
+    raise error_class(parsed_response, operation_name)
```
Comment (hannes-ucsc) [inline README.md]: Replace with `...` like we do in doctest assertions.

**Comment by hannes-ucsc** [comment]
@achave11 note https://github.com/DataBiosphere/azul/pull/1369/commits/e7c2df00aeab985187b3e89ff79ee64c1e4d6e97

**Comment by achave11** [comment]
@hannes-ucsc when you force push to this branch, changes to the lambda policy for service were not captured.

**Comment by hannes-ucsc** [comment]
I forgot to pull. Fixed now.

**Pull request merged**